src/commands/game_options/groups.ts

```diff
@@ -235,4 +257,101 @@ export default class GroupsCommand implements BaseCommand {
+     * @param interaction - The completed groups interaction
+     * @param messageContext - The source of the interaction
+     */
+    static async processGroupsChatInputInteraction(
+        interaction: Eris.CommandInteraction,
+        messageContext: MessageContext
+    ): Promise<void> {
+        if (interaction instanceof Eris.CommandInteraction) {
+            if (
+                interaction.data.type ===
+                Eris.Constants.ApplicationCommandTypes.CHAT_INPUT
+            ) {
+                logger.info(
+                    `${getDebugLogHeader(interaction)} | ${
+                        interaction.data.name
+                    } slash command received`
+                );
+
+                const groups: Array<MatchedArtist> = _.uniqBy(
+                    interaction.data.options.map(
+                        (x) => JSON.parse(x["value"]) as MatchedArtist
+                    ),
+                    "id"
+                );
+
+                const guildPreference =
+                    await GuildPreference.getGuildPreference(
+                        interaction.guildID
+                    );
+
+                await guildPreference.setGroups(groups);
+                tryCreateInteractionSuccessAcknowledgement(
+                    interaction,
+                    LocalizationManager.localizer.translate(
+                        interaction.guildID,
+                        "command.groups.interaction.groupsUpdated.title"
+                    ),
+                    LocalizationManager.localizer.translate(
+                        interaction.guildID,
+                        "command.groups.interaction.groupsUpdated.description"
+                    )
+                );
+
+                await sendOptionsMessage(
+                    Session.getSession(messageContext.guildID),
+                    messageContext,
+                    guildPreference,
+                    [{ option: GameOption.GROUPS, reset: false }]
+                );
+            }
+        }
+    }
+
+    /**
+     * Handles showing suggested artists as the user types for the groups slash command
+     * @param interaction - The interaction with intermediate typing state
+     */
+    static processGroupsAutocompleteInteraction(
+        interaction: Eris.AutocompleteInteraction
+    ): void {
+        const userInput = interaction.data.options.filter(
+            (x) => x["focused"]
+        )[0]["value"] as string;
+
+        const artistEntryToInteraction = (
+            x: MatchedArtist
+        ): { name: string; value: string } => ({
+            name: x.name,
+            value: JSON.stringify(x),
+        });
+
+        if (userInput === "") {
+            tryAutocompleteInteractionAcknowledge(
+                interaction,
+                State.topArtists.map((x) => artistEntryToInteraction(x))
+            );
+
+            return;
+        }
+
+        const matchingGroups = Object.entries(State.artistToEntry)
```

Comment:
Should we filter groups they've already inputted?

Reply:
I don't think the currently entered groups are accessible from this interaction